vars/deployClusterPipeline.groovy

```diff
@@ -0,0 +1,198 @@
+        steps {
+          script {
+            def currentProject = "${env.PROJECT_NAME}"
+            def setPermissions = {
+              openshift.withProject("images-${this._buildName}") {
+                openshift.raw("policy", "add-role-to-group", "admin", "system:serviceaccounts:${currentProject}")
+                openshift.raw("policy", "add-role-to-user", "view", "system:anonymous")
+                openshift.raw("policy", "add-role-to-user", "system:image-puller", "system:anonymous")
+              }
+            }
+            openshift.withCluster() {
+              if ("${params.PROVISIONER_TOKEN}".length() > 0) {
+                openshift.doAs("${params.PROVISIONER_TOKEN}") {
+                  openshift.newProject("images-${this._buildName}")
+                  setPermissions()
+                }
+              } else {
+                openshift.newProject("images-${this._buildName}")
+                setPermissions()
+              }
+            }
+          }
+        }
+      }
+      stage("Tag Images") {
+        when { expression { return this._imageBase ? false : true } }
+        steps {
+          script {
+            def currentProject = "${env.PROJECT_NAME}"
+            openshift.withCluster() {
+              openshift.withProject("${stableImageNamespace}") {
+                openshift.selector("is").withEach {
+                  def name = it.object().metadata.name
+                
+                  def overrideTag = tagOverrides[name]
+                  if (overrideTag) {
+                    openshift.withProject("images-${this._buildName}") {
+                      openshift.tag("${stableImageNamespace}//${name}:${overrideTag}", "${name}:ci")
+                    }
+                  }
+                  else {
+                    def tags = openshift.selector("is", "${this._buildName}").object().status.tags
+                    for (i = 0; i < tags.size(); i++) {
+                      // Only tag an image stream if it contains the stableImageTag we're looking for
+                      if (tags[i].tag == "${stableImageTag}") {
+                        openshift.withProject("images-${this._buildName}") {
+                          openshift.tag("${stableImageNamespace}/${name}:${stableImageTag}", "${name}:ci")
+                        }
+                      }
+                    }
+                  }
+                }
+              }
+              // Apply tags resulting from current build
+              def tags = openshift.selector("is", "${this._buildName}").object().status.tags
+              for (i = 0; i < tags.size(); i++) {
+                def tag = tags[i].tag
+                openshift.withProject("images-${this._buildName}") {
+                  openshift.tag("${currentProject}/${this._buildName}:${tag}", "${tag}:ci")
+                }
+              }
+
+              // Get the image-base for origin install
```

Comment:
Unclear as to what this is

Reply:
As mentioned above it's the ref to use for deriving all origin images. It's used here:
https://github.com/openshift/release/blob/master/cluster/ci/origin/pipelines/deploy/Jenkinsfile#L89